### PR TITLE
`Exam mode`: Fix a bug that prevented the exam detail page from displaying anything after a reset

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

If an instructor resets an exam, the exam details disappear and they have to reload the page to continue working:

grafik

### Description

- Set course access rights on the exam returned by the server after reset so the authorization check succeeds and details are shown again
- Clearing an internal array of the exercise group list on change so it's freshly filled, otherwise the exercise groups of the reset exam would be appended
- Added an success alert for a successful reset as it's not super clear without it

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with a few registered students

1. Generate student exams for the students
2. Prepare exercise start
3. Go to exam management view of the exam (with the checklist)
4. Verify that all exercise group in the exercise group list are listed correctly and have a number of participations
grafik
5. Reset the exam
6. Verify that you get a success alert after you reset it and the modal closed
7. Verify that you can still see everything without reloading on the checklist and **don't** get an empty page like in the first image of this PR
8. Verify that the exercise group table does not contain duplicated exercises, and that the participants column shows nothing / only "-"

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-management.service.ts | 73% | 95% |
| exam-detail.component.ts | 75% | 96% |
| exam-checklist-exercisegroup-table.component.ts | 76% | 94% |

### CODE REVIEW

File: src/main/webapp/app/exam/manage/exam-management.service.ts [hunk 1]
```diff
@@ -403,7 +403,14 @@ export class ExamManagementService {
      * @param examId The exam id.
      */
     reset(courseId: number, examId: number): Observable<EntityResponseType> {
-        return this.http.delete<Exam>(`${this.resourceUrl}/${courseId}/exams/${examId}/reset`, { observe: 'response' });
+        return this.http.delete<Exam>(`${this.resourceUrl}/${courseId}/exams/${examId}/reset`, { observe: 'response' }).pipe(
+            map((res: EntityResponseType) => ExamManagementService.convertCourseResponseDateFromServer(res)),
+            tap((res: EntityResponseType) => {
+                if (res.body?.course) {
+                    this.accountService.setAccessRightsForCourse(res.body.course);
+                }
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exam/manage/exam-management.service.ts:411]: Should we maybe already include this in the convertCourseResponse method?

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me, just one thought:

File: src/main/webapp/app/exam/manage/exam-management.service.ts (hunk 1 above)
Comment (bassner) [reply src/main/webapp/app/exam/manage/exam-management.service.ts]: Yeah, we can. Check out 9c029e0

**Review by xlieA — COMMENTED** [review]
Tested on ts1. Everything described in the testing steps works.
I don't know if this is related with this PR, but when I go to Course Management -> *Course name* -> Exam Management, I get this error and I can't see any of the previously create exams nor the one I just created
Screenshot 2022-07-29 205450
.

**Comment by bassner** [comment]
@xlieA Did you test build #4 or build #5 on Bamboo? This is probably related to why the test failed as well and was fixed in 4cce253

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good!

**Review by xlieA — APPROVED** [review]
Retested on ts1. Now everything works, there is no error message anymore.

**Review by canberkanar — APPROVED** [review]
Tested on TS1 according to the instruction. All is well.

**Review by akesfeden — APPROVED** [review]
Tested on TS2, reset shows a success message and details are visible without a page reload, "Participants" are set to "-" as expected.

**Pull request merged**